components/motor/ulnstepper/28byj-48.go

```diff
@@ -391,13 +399,16 @@ func (m *uln28byj) SetPower(ctx context.Context, powerPct float64, extra map[str
 		m.logger.CWarn(ctx, warning)
 		if err != nil {
 			m.logger.CError(ctx, err)
+			// only stop if we receive a zero RPM error
+			return m.Stop(ctx, extra)
 		}
-		return m.Stop(ctx, extra)
 	}
 
 	m.lock.Lock()
 	defer m.lock.Unlock()
-	m.targetStepPosition = int64(math.Inf(int(powerPct)))
+	direction := motor.GetSign(powerPct) // get the direction to set target to -ve/+ve Inf
+	m.targetStepPosition = int64(math.Inf(int(direction)))
```

Comment:
I'm pretty sure `math.Inf` already does this. like it sets infinity +Inf for a positive number and -Inf for a negative number

Reply:
point being, you shouldn't need to call `GetSign`

Reply:
It was not doing so, I wonder if the problem was within int then...

ah that must be it, -0.2 powerpct is zero ints and that gets passed at positive to `Inf`

Reply:
It was only going in one direction before, so it was wrong.

Reply:
ah got it that makes sense. this works then